### PR TITLE
initial nokogiri libgumbo integration

### PR DESCRIPTION
**Pull request by fuzzy-boiii23a**
criticality score: 0.75414

approval: https://github.com/sparklemotion/nokogiri/discussions/2992

Please let me know if there is anything else that needs changing and just to note i'll be creating a pull request to integrate the harness into the upstream nokogiri repo sometime this week.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
We approve of adding thsi project. You just need to add a real user first.

File: projects/nokogiri/project.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+homepage: "nokogiri.org"
+language: c
+primary_contact: "nokogiri-oss-fuzz@googlegroups.com"
```
[the remark this answers is not in the document]
Comment (fuzzy-boiii23a) [reply projects/nokogiri/project.yaml]: Done, sorry for that will take note not to do this in the future

File: projects/nokogiri/gumbo.dict [hunk 2]
```diff
@@ -0,0 +1,560 @@
+#
```
Comment (jonathanmetzman) [inline projects/nokogiri/gumbo.dict:1]: Can you please put corpus.zip somewhere else and download it. We don't want binary files in our repo.

File: projects/nokogiri/project.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+main_repo: 'https://github.com/sparklemotion/nokogiri.git'
+sanitizers:
+  - address
+  - undefined
+architectures:
+  - x86_64
+fuzzing_engines:
+  - afl
+  - honggfuzz
+  - libfuzzer
```
Comment (jonathanmetzman) [inline projects/nokogiri/project.yaml:15]: centipede was breaking?

File: projects/nokogiri/project.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+language: c
+primary_contact: "fuzzy.boiiii23a@gmail.com"
+auto_ccs:
+  - mike.dalessio@gmail.com
+main_repo: 'https://github.com/sparklemotion/nokogiri.git'
+sanitizers:
+  - address
+  - undefined
+architectures:
+  - x86_64
```
Comment (jonathanmetzman) [inline projects/nokogiri/project.yaml:11]: Let's omit this and leave it implicit.

**Review by jonathanmetzman — COMMENTED** [review]
Everything looks good

File: projects/nokogiri/project.yaml (hunk 4 above)
Comment (fuzzy-boiii23a) [reply projects/nokogiri/project.yaml]: done

**Comment by fuzzy-boiii23a** [comment]
i resolved your comments @jonathanmetzman , please let me know if there are any more improvements i should make, thanks for the help it's much appreciated

**Pull request merged**